wrappers/java/src/main/java/org/hyperledger/indy/sdk/agent/Agent.java

```diff
@@ -0,0 +1,329 @@
+package org.hyperledger.indy.sdk.agent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+import org.hyperledger.indy.sdk.LibSovrin;
+import org.hyperledger.indy.sdk.SovrinException;
+import org.hyperledger.indy.sdk.SovrinJava;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentAddIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseConnectionResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentCloseListenerResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentConnectResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentListenResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentRemoveIdentityResult;
+import org.hyperledger.indy.sdk.agent.AgentResults.AgentSendResult;
+import org.hyperledger.indy.sdk.pool.Pool;
+import org.hyperledger.indy.sdk.wallet.Wallet;
+
+import com.sun.jna.Callback;
+
+/**
+ * agent.rs API
+ */
+public class Agent extends SovrinJava.API {
+
+	private Agent() {
+
+	}
+
+	/*
+	 * STATIC METHODS
+	 */
+
+	public static Future<AgentConnectResult> agentConnect(
+			Pool pool,
+			Wallet wallet,
+			String senderDid,
+			String receiverDid,
+			Callback messageCb) throws SovrinException {
+
+		final CompletableFuture<AgentConnectResult> future = new CompletableFuture<> ();
+
+		Callback connectionCb = new Callback() {
```

Comment:
> connectionCb = new Callback()

We are creating a new instance of Callback for each call and perform passing of this callback to c function. JVM can't manage reference counting in C code. As result, GC will free this object on the next GC iteration and it will cause bad behavior like crashes or missed callback calls. 

To avoid this I suggest the following:

1. For each call create one static callback instance and `static Map<i32 /* command handle */, Future>`
2. Use global atomic counter to generate new command handles
3. When user calls our method generate new command handles and put corresponded Future to static Map
4. When libsovrin calls our callback resolve corresponded future with result